### PR TITLE
adding elasticsearch stuff

### PR DESCRIPTION
**Pull request by malkolm**

### CODE REVIEW

**Comment by chrisgilbert** [comment]
Looks good Malcolm.  :+1:

File: tyr/servers/elasticsearch/server.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    IAM_ROLE_POLICIES = [
+        'allow-describe-tags',
+        'allow-describe-instances',
+        'allow-volume-control',
+        'allow-update-route53-{environment}'
+    ]
+
+    def __init__(self, group=None, server_type=None, instance_type=None,
+                    environment=None, ami=None, region=None, role=None,
```
Comment (citruspi) [inline tyr/servers/elasticsearch/server.py]: These are indented a little too far - they should align with `self`.

File: tyr/servers/elasticsearch/server.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+                    keypair=None, availability_zone=None, security_groups=None,
+                    block_devices=None, chef_path=None, subnet_id=None,
+                    dns_zones=None):
+
+        if server_type is None: server_type = self.SERVER_TYPE
+
+        super(ElasticsearchServer, self).__init__(group, server_type, instance_type,
```
Comment (citruspi) [inline tyr/servers/elasticsearch/server.py]: This line goes over 79 characters.

File: tyr/servers/elasticsearch/server.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+                                             environment, ami, region, role,
```
Comment (citruspi) [inline tyr/servers/elasticsearch/server.py]: These should be indented a little further - they should align with `group`.

File: tyr/servers/elasticsearch/server.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+    IAM_ROLE_POLICIES = [
+        'allow-describe-tags',
+        'allow-describe-instances',
+        'allow-volume-control',
+        'allow-update-route53-{environment}'
+    ]
+
+    def __init__(self, group=None, server_type=None, instance_type=None,
+                environment=None, ami=None, region=None, role=None,
```
Comment (citruspi) [inline tyr/servers/elasticsearch/server.py]: These are a little under-indented.

File: tyr/servers/elasticsearch/server.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+                keypair=None, availability_zone=None, security_groups=None,
+                block_devices=None, chef_path=None, subnet_id=None,
+                dns_zones=None):
+
+        if server_type is None: server_type = self.SERVER_TYPE
+
+        super(ElasticsearchServer, self).__init__(group, server_type,
+                                                instance_type, environment,
```
Comment (citruspi) [inline tyr/servers/elasticsearch/server.py]: These are a little under-indented.

**Comment by citruspi** [comment]
This looks good to me, but we should probably hold off on merging this in until #54 is merged in - it won't work until then.

**Pull request merged**